src/core/bpf-firewall.c

```diff
@@ -914,10 +914,7 @@ int bpf_firewall_supported(void) {
 void emit_bpf_firewall_warning(Unit *u) {
         static bool warned = false;
 
-        assert(u);
-        assert(u->manager);
-
-        if (!warned && !MANAGER_IS_TEST_RUN(u->manager)) {
```

Comment:
given that this is called from a very different place I'd actually keep the nicely encapsulated check here in place I think.